llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -768,7 +776,10 @@ struct AddressSanitizer {
   ShadowMapping Mapping;
   FunctionCallee AsanHandleNoReturnFunc;
   FunctionCallee AsanPtrCmpFunction, AsanPtrSubFunction;
+  FunctionCallee AsanSetShadowDeviceLocalFunc;
   Constant *AsanShadowGlobal;
+  Constant *AsanShadowDevicePrivate;
+  std::unordered_map<std::string, GlobalVariable *> GlobalStringMap;
```

Comment:
Why not [`StringMap`](https://llvm.org/doxygen/classllvm_1_1StringMap.html)? 

[LLVM Programmer's Manual](https://llvm.org/docs/ProgrammersManual.html#other-map-like-container-options) says that core LLVM components never use `unordered_map`, because it is too expensive

Reply:
Thanks for you reminder!